### PR TITLE
Adding openmm-nonbonded-slicing

### PR DESCRIPTION
**Pull request by craabreu**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by craabreu** [comment]
Hi @jananzhu, I'm adding you as a maintainer here, ok?

**Comment by jananzhu** [comment]
@craabreu Sounds good, I'm willing to be a maintainer for this package.

**Comment by craabreu** [comment]
@conda-forge/staged-recipes, ready for review

@conda-forge-admin, please ping team @conda-forge/staged-recipes

**Comment by craabreu** [comment]
@conda-forge/staged-recipes or @conda-forge/help-python-c, would you please review this PR?

@conda-forge-admin, please ping team @conda-forge/staged-recipes

@conda-forge-admin, please ping team @conda-forge/help-python-c

**Comment by craabreu** [comment]
@conda-forge-admin, please ping team @conda-forge/help-python-c

**Comment by craabreu** [comment]
@conda-forge-admin, please ping team @conda-forge/help-python

**Comment by craabreu** [comment]
Could you please review this recipe? It passed all tests on Jan 5.

File: recipes/openmm-nonbonded-slicing/build.sh [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+if [[ "$target_platform" == osx* ]]; then
+    CMAKE_FLAGS+=" -DCMAKE_OSX_SYSROOT=${CONDA_BUILD_SYSROOT}"
+    CMAKE_FLAGS+=" -DCMAKE_OSX_DEPLOYMENT_TARGET=${MACOSX_DEPLOYMENT_TARGET}"
+fi
+
+# Build in subdirectory and install.
+mkdir -p build
+cd build
+cmake ${CMAKE_FLAGS} ${SRC_DIR}
```
Comment (xhochy) [inline recipes/openmm-nonbonded-slicing/build.sh:33]:
```suggestion
cmake ${CMAKE_ARGS} ${CMAKE_FLAGS} ${SRC_DIR}
```

File: recipes/openmm-nonbonded-slicing/build.sh [hunk 2]
```diff
@@ -0,0 +1,45 @@
+#!/bin/bash
+
+set -euxo pipefail
+
+rm -rf build || true
+
+CMAKE_FLAGS="  -DCMAKE_INSTALL_PREFIX=${PREFIX}"
```
Comment (xhochy) [inline recipes/openmm-nonbonded-slicing/build.sh:7]: By using the provided CMAKE_ARGS you should be able to get rid of most of the flags here.

File: recipes/openmm-nonbonded-slicing/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,79 @@
+{% set user = "craabreu" %}
```
Comment (xhochy) [inline recipes/openmm-nonbonded-slicing/meta.yaml:1]: Don't template this but write it plain.

File: recipes/openmm-nonbonded-slicing/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+  url: https://github.com/{{ user }}/{{ name }}/archive/v{{ version }}.tar.gz
+  sha256: 796e504837d27e747ded1f55a3bd01f8bef4c72036dd497c473d1579220521b2
+  patches:
+  - cmakelists.patch
+
+build:
+  number: 0
+  skip: true  # [win or osx or (linux and cuda_compiler_version in (undefined, 'None', '10.2'))]
+  # Note: osx to be included as soon as OpenMM 8.0 is released
+  missing_dso_whitelist:
```
Comment (xhochy) [inline recipes/openmm-nonbonded-slicing/meta.yaml:19]: Should no longer be necessary.

File: recipes/openmm-nonbonded-slicing/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}  # [linux and cuda_compiler_version != 'None']
+    - {{ cdt('mesa-libgl-devel') }}  # [linux]
+    - cmake
+    - make
+    - conda-build
```
Comment (xhochy) [inline recipes/openmm-nonbonded-slicing/meta.yaml:30]: You have this as you are calling the recipe with it.
```suggestion
```

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipes/openmm-nonbonded-slicing/meta.yaml (hunk 3 above)
Comment (craabreu) [reply recipes/openmm-nonbonded-slicing/meta.yaml]: Done.

**Review by craabreu — COMMENTED** [review]
Thank you, @xhochy, for reviewing. I followed the suggestions as closely as possible. I had to update the recipe and building script to reflect changes in the package. I also updated the recipe following the latest version of the `openmm-torch` feedstock.

**Review by xhochy — APPROVED** [review]

**Pull request merged**